main/src/ca/uwaterloo/flix/runtime/interpreter/Channel.java

```diff
@@ -48,11 +48,20 @@
    */
   private int bufferSize;
 
+  /**
+   * a flag for whether the channel is unbuffered or not
+   */
+  private boolean unbuffered = false;
```

Comment:
can this be final?

Reply:
We change it in the constructor if we're given a size of 0 so i don't think so?

Reply:
We've made a new commit changing this to final